src/ComputeEngineImplementation.java

```diff
@@ -43,6 +58,11 @@ public EngineResponse submitRequestHelper(UserRequest userRequest) throws Except
 
 	requestResult.setResultString(processResultString(userRequest, (ArrayList<Integer>) resultStream.getIntegers()));
 
+  if (internalRequest) {
+    dataStoreAPI.setOutputList(new ArrayList<>(List.of(requestResult.getResultString())));
+    dataStoreAPI.writeOutput(new OutputRequest(userRequest.getUserRequestDestination().outputPath));

```

Comment:
Modify for `FileUserRequestDestination`:
```suggestion
    UserRequestDestination requestDestination = userRequest.getUserRequestDestination();
    if (requestDestination instanceof FileUserRequestDestination fileRequestDestination){
        dataStoreAPI.writeOutput(new OutputRequest(fileRequestDestination.getFileName()));
    }
```